standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```
@@ -566,15 +600,15 @@ default void dropTable(String catName, String dbName, String tableName)
    * @throws TException Thrift transport exception
    */
   @Deprecated
-  void truncateTable(String dbName, String tableName, List<String> partNames) throws MetaException, TException;
+  default void truncateTable(String dbName, String tableName, List<String> partNames) throws MetaException, TException{}
 
-  void truncateTable(TableName table, List<String> partNames) throws TException;
+  default void truncateTable(TableName table, List<String> partNames) throws TException{}
```

should throw UnsupportedOperationException